Improve build automation

lahma opened the pull request:
* fix workflow definitions (this repo uses `master` instead of `main`, pr worfklow should track PRs and build workflow pushes/tags)
* add minimal .gitattributes (fixes build.sh)
* fix Acornima.sln
* exclude macOS from recursion testing

After changing GitHub definitions for NUKE, I needed to run `build.cmd` to generate new YAML files, that's the procedure for such changes.

adams85 commented:
Thanks for the fixes! BTW, how about this: https://github.com/adams85/acornima/pull/2#discussion_r1490209180

lahma commented:
> Thanks for the fixes! BTW, how about this: [#2 (comment)](https://github.com/adams85/acornima/pull/2#discussion_r1490209180)

Good catch, fixed the logic and removed some redundancy. Not sure how the actual version will determined, was just a templete, one possibility is GitVersion tasks and automation.
